larndsim/quenching.py

```diff
@@ -47,3 +47,5 @@ def quench(tracks, mode):
             raise RuntimeError("Invalid recombination value")
 
         tracks[itrk]["n_electrons"] = recomb * dE * consts.MeVToElectrons
+
+        light_dat[itrk,0]["n_photons_edep"] = (dE/consts.Wph - tracks[itrk]["n_electrons"])*consts.ScintPreScale
```

Comment:
what exactly is the shape of `light_dat`? Why do you need to do `light_dat[itrk,0]`?

Reply:
`light_dat` right now holds both the number of created photons and the  incident photons per optical channel per edep (shape is `(n_edeps, n_channels)`).  Ideally, we want the number of created photons (which _should_ have shape `(n_edeps)`) to be a part of the tracks database (in the same way as `n_electrons` is currently handled).  `n_electrons` seems to be added but not populated in a previous step in the simulation chain, so it's not clear how to include `n_photons` in this way.

Reply:
As it is written, we are just using the first channel to store the `n_photons` data.  It would be better for us to store this in another 1-dimensional dataset for now, and eventually add `n_photons` to the `tracks` set.

Reply:
`n_electrons` is added at `dumpTree.py` stage. It's not ideal but it seems the path of least resistance. Otherwise we could think of including an extra step in `simulate_pixels.py` where we add the fields we need (`n_electrons`, `pixel_plane`, etc.).